doc/neps/roadmap.rst

```diff
@@ -6,110 +6,112 @@ This is a live snapshot of tasks and features we will be investing resources
+as many libraries that extend NumPy's model in various ways.  Work in NumPy to
+facilitate interoperability with all such packages, and the code that uses them,
+may include (among other things) interoperability protocols, better duck typing
+support and ndarray subclass handling.
+
+- The ``__array_function__`` protocol is currently experimental and needs to be
+  matured. See `NEP 18`_ for details.
+- New protocols for overriding other functionality in NumPy may be needed.
+- Array duck typing, or handling "duck arrays", needs improvements.  See
+  `NEP 22`_ for details.
+
+Extensibility
+-------------
 
-- Mixins like `NDArrayOperatorsMixin`:
+We aim to make it much easier to extend NumPy. The primary topic here is to
+improve the dtype system.
 
-  - for mutable arrays
-  - for reduction methods implemented as ufuncs
+- Easier custom dtypes:
 
-Better dtypes
--------------
-
-- Easier custom dtypes
   - Simplify and/or wrap the current C-API
   - More consistent support for dtype metadata
   - Support for writing a dtype in Python
+
 - New string dtype(s):
-  - Encoded strings with fixed-width storage (utf8, latin1, ...) and/or
-  - Variable length strings (could share implementation with dtype=object, but are explicitly type-checked)
-  - One of these should probably be the default for text data. The current behavior on Python 3 is neither efficient nor user friendly.
-- `np.int` should not be platform dependent
-- better coercion for string + number
 
-Random number generation policy & rewrite
------------------------------------------
+  - Encoded strings with fixed-width storage (utf8, latin1, ...) and/or
+  - Variable length strings (could share implementation with dtype=object,
+    but are explicitly type-checked)
+  - One of these should probably be the default for text data. The current
+    behavior on Python 3 is neither efficient nor user friendly.
 
-`NEP 19`_ and a `reference implementation`_
+- `np.int` should not be platform dependent
+- Better coercion for string + number
 
-Indexing
---------
+Performance
+-----------
 
-vindex/oindex `NEP 21`_
+We want to further improve NumPy's performance, through:
 
-Infrastructure
---------------
+- Better use of SIMD instructions, also on platforms other than x86.
+- Reducing ufunc overhead.
+- Optimizations in individual functions.
 
-NumPy is much more than just the code base itself, we also maintain
-docs, CI, benchmarks, etc.
+Furthermore we would like to improve the benchmarking system, in terms of coverage,
+easy of use, and publication of the results as part of the docs or website:
 
-- Rewrite numpy.org
-- Benchmarking: improve the extent of the existing suite, and run & render
-  the results as part of the docs or website.
+- Hardware: find a machine that can reliably run serial benchmarks
+- ASV produces graphs, could we set up a site? Currently at
+  https://pv.github.io/numpy-bench/, should that become a community resource?
```

Comment:
If this document also aims at funding, it seems more like a question and less like a Roadmap item? Maybe do.

... website. Specifically:
- Improve the benchmarking infrastructure with new hardware.
- Publish Benchmarking more centrally both for users and developers.

anyway, just a thought.

Reply:
It doesn't directly aim at funding. But yeah that could be fixed up a bit.

Reply:
done